src/Symfony/Component/HttpKernel/UriSigner.php

```diff
@@ -106,4 +134,22 @@ private function buildUrl(array $url, array $params = []): string
 
         return $scheme.$user.$pass.$host.$port.$path.$query.$fragment;
     }
+
+    private function getExpirationTime(\DateTimeInterface|\DateInterval|int $expiration): string
+    {
+        if ($expiration instanceof \DateTimeInterface) {
+            return $expiration->format('U');
+        }
+
+        if ($expiration instanceof \DateInterval) {
+            return (new \DateTimeImmutable())->add($expiration)->format('U');
+        }
+
+        return (string) $expiration;
+    }
+
+    private function isExpired(int $expiration): bool
+    {
+        return ((int) microtime(true)) >= $expiration;
```

Comment:
use `time()` rather than `microtime()` if you don't care about sub-second precision. It is better than doing a precision-losing cast IMO.